### ISSUE TITLE
False positive garbage value from a constructor with inheritance

### ISSUE THREAD

**Issue by chfast** [llvm/llvm-project#59493]
After constructing `A` with default constructor Clang Analyzer thinks the `a.bytes` contains a garbage value. This can later cause reporting warnings like:
- Assigned value is garbage or undefined
- Undefined or garbage value returned to caller
- The left operand of [...] is a garbage value

https://godbolt.org/z/3cerPT64a

```cpp
struct B {
    unsigned char bytes[1];
};

struct A : B {
    explicit A() : B{{0}} {}
};

auto test1(unsigned char& o) {
    A a;
    o = a.bytes[0];
}
```

**Comment by steakhal** [comment]
@isuckatcs maybe you could have a look, it seems like it could be related to init loops.

**Comment by isuckatcs** [comment]
At first glance I would say this has something to do with the `B{...}` initialization. If you replace the array by a single variable, the same false positive is reported. See it on [godbolt](https://godbolt.org/z/vEvfY67Yq).

Also not that if in the original snippet you use `B(...)` instead, the false positives are not reported. See it again on [godbolt](https://godbolt.org/z/K6ab51G8a).

IIRC we've seen something similar in the past and @haoNoQ  mentioned that a new construction context needs to be created to solve this problem.

**Comment by steakhal** [comment]
Fixed by 39743580aff515ba47cb7f9ffb2b5c538cff91da

**Comment by Snape3058** [comment]
A test case has been added covering this issue in https://github.com/llvm/llvm-project/pull/71073
